.travis.yml

```diff
@@ -10,6 +16,13 @@ matrix:
   exclude:
     - os: osx
       compiler: gcc
+before_install:
+        - apt-get source libnl-dev && cd libnl* && ./configure --prefix=$HOME && make && make install
+        # Using latest libibverbs and rdmacm as the system provided versions are old
+        - git clone https://git.kernel.org/pub/scm/libs/infiniband/libibverbs.git
+        - cd libibverbs && ./autogen.sh && ./configure --prefix=$HOME && make && make install
+        - git clone git://flatbed.openfabrics.org/~shefty/librdmacm.git
+        - cd librdmacm && ./autogen.sh && ./configure --prefix=$HOME CC=gcc && make && make install
```

Comment:
this section too  should be conditional based on TRAVIS_OS_NAME.  It might be easier to add a script in libfabric which is invoked conditioned upon the os-type.